packages/block-editor/src/components/link-control/index.js

```diff
@@ -131,16 +131,25 @@ export function LinkControl( {
 	}, [ handleDirectEntry, fetchSearchSuggestions ] );
 
 	// Render Components
-	const renderSearchResults = ( { suggestionsListProps, buildSuggestionItemProps, suggestions, selectedSuggestion, isLoading } ) => {
+	const renderSearchResults = ( { suggestionsListProps, buildSuggestionItemProps, suggestions, selectedSuggestion, isLoading, isInitialSuggestions } ) => {
 		const resultsListClasses = classnames( 'block-editor-link-control__search-results', {
 			'is-loading': isLoading,
 		} );
 
 		const manualLinkEntryTypes = [ 'url', 'mailto', 'tel', 'internal' ];
+		const searchResultsLabelId = `block-editor-link-control-search-results-label-${ instanceId }`;
+		const labelText = isInitialSuggestions ? __( 'Recently modified' ) : sprintf( __( 'Search results for %s' ), inputValue );
+		const SearchResultsLabel = (
+			<span className="block-editor-link-control__search-results-label" id={ searchResultsLabelId }>
+				{ labelText }
+			</span>
+		);
 
 		return (
 			<div className="block-editor-link-control__search-results-wrapper">
-				<div { ...suggestionsListProps } className={ resultsListClasses }>
+				{ ! isInitialSuggestions ? <VisuallyHidden>{ SearchResultsLabel }</VisuallyHidden> : SearchResultsLabel }
+
+				<div { ...suggestionsListProps } className={ resultsListClasses } aria-labelledby={ searchResultsLabelId }>
```

Comment:
Nice job on adding `aria-labelledby`! 🙌 

I think there may be a slight difference on using that on visible vs hidden labels: https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/listbox_role

I'm not positive I'm fully understanding the implementation though: 
> **aria-label**
A human-readable string value which identifies the listbox.  If there's a visible label, then aria-labelledby should be used instead to refer to that label.

> **aria-labelledby**
Identifies the visible element or elements in a space-separated list of element IDs which identify the listbox.  If there's no visible label, then aria-label should be used instead to include a label.  (Note: "labelled", with two L's, is the correct spelling based on the accessibility API conventions.)

Reply:
Good catch but I don't follow the a11y guidelines here. Do I need to just add `aria-label` on the element when it's not visible?